drivers/counter/counter_mcux_rtc.c

```diff
@@ -236,6 +237,15 @@ static int mcux_rtc_init(const struct device *dev)
 	RTC_GetDefaultConfig(&rtc_config);
 	RTC_Init(config->base, &rtc_config);
 
+	if (config->rtc_clock_source) {
+#if (defined(FSL_FEATURE_RTC_HAS_LPO_ADJUST) && FSL_FEATURE_RTC_HAS_LPO_ADJUST)
+		/* The RTC prescaler increments using the LPO 1 kHz clock
+		 * instead of the RTC clock
```

Comment:
instead of adding this code which will do the opposite of what was configured in DT, maybe use a BUILD_ASSERT to check for this case and cause a build error to inform the user that you cannot choose the RTC clock for that instance of the RTC

Reply:
Sorry, I don't understand what the BUILD_ASSERT restrictions are here. 
The LPO clock can only be enabled if `clock-source = "LPO"` is set and this macro is true.

Reply:
ok, I see, I was confused by the code because I thought the boolean "rtc_clock_source" would be true if the RTC is the clock source chosen and not the LTO. I think you should change this so that it is an enum instead of a boolean for readability.

Reply:
I think the build assert can be useful to check when LPO is chosen as clock source AND is supported by the device

Reply:
I think that's what I meant too when I first made the comment, so ditto

I have too many PRs to review I didnt even remember what my own review means, so I guess this means I need to leave better reviews

Reply:
Updated.